### ISSUE TITLE
dotnet test does not find any tests if environment variable is set that ends with .dll or .exe

### ISSUE THREAD

**Issue by jbe2277** [dotnet/sdk#40848]
We use environment variables in UI tests to specify on our build machines where our application (executable) can be found.

`dotnet test UITests/UITests.sln -e UITestExe="C:\OurApp.exe"`

By defining an environment variable that ends on `.exe` or `.dll` the `dotnet test` does not find unit tests anymore.

Reason: In this case dotnet test searches the file specified as environment variable for unit tests.

- Related issue: #28592 
  - Just an ending with `dll` works but `.dll` does not work
- PR that solved the previous issue: #28662

It seems the the PR #28662 did not solve this completely.

Version:
```
.NET SDK:
 Version:           8.0.204
 Commit:            c338c7548c
 Workload version:  8.0.200-manifests.7d36c14f
```

**Comment by jbe2277** [comment]
Yes, this works: `dotnet [env:UITestExe="C:\OurApp.exe"] test UITests/UITests.sln`

I did not find any information about this special syntax. Is this somewhere documented?

**Comment by WeihanLi** [comment]
The env directive is support by default by `System.CommandLine`

**Comment by nohwnd** [comment]
This is a problem of how we parse the parameters to decide if we should call msbuild or vstest.console directly.
